src/sagemaker/estimator.py

```diff
@@ -78,7 +79,17 @@ def __init__(self, role, train_instance_count, train_instance_type,
         self.train_volume_size = train_volume_size
         self.train_max_run = train_max_run
         self.input_mode = input_mode
-        self.sagemaker_session = sagemaker_session or Session()
+
+        if self.train_instance_type in ('local', 'local_gpu'):
+            self.local_mode = True
+            if self.train_instance_type == 'local_gpu' and self.train_instance_count > 1:
```

Comment:
Create an issue to support distributed local gpu in future, but that's going to be tough to do if the local machine only has a single gpu.